documentation/development-docs/azure-powershell-design-guidelines.md

```diff
@@ -0,0 +1,308 @@
+
+> _Each parameter set must have at least one unique parameter. Good cmdlet design indicates this unique parameter should also be mandatory if possible. If your cmdlet is designed to be run without parameters, the unique parameter cannot be mandatory._
+
+#### Positional Parameters Limit
+
+It is possibile to call a PowerShell cmdlet without providing the parameter names, but just the values you would like to pass through. This is done by specifying the position at which the value of each parameter should be provided by using the `Position` property for a parameter.  However, when there are too many positional parameters in a single parameter set, it can be difficult for the user to remember the exact ordering in which the parameter values should be provided. From the remarks section of [Parameter Attribute Declaration](https://msdn.microsoft.com/en-us/library/ms714348(v=vs.85).aspx):
+
+> _When you specify positional parameters, limit the number of positional parameters in a parameter set to less than five. And, positional parameters do not have to be contiguous. Positions 5, 100, and 250 work the same as positions 0, 1, and 2._ 
+
+In addition, there should be no two parameters with the same position in the same parameter set. From the remarks section of [Parameter Attribute Declaration](https://msdn.microsoft.com/en-us/library/ms714348(v=vs.85).aspx):
+
+> _No parameter set should contain more than one positional parameter with the same position._
+
+#### ValueFromPipeline Limit
+
+Allowing the user to pipe an object from one cmdlet to another is a major scenario in PowerShell, but allowing multiple parameters in the same parameter set to accept their value from the pipeline can cause issues. From the remarks section of [Parameter Attribute Declaration](https://msdn.microsoft.com/en-us/library/ms714348(v=vs.85).aspx):
+
+> _Only one parameter in a parameter set should declare ValueFromPipeline = true. Multiple parameters can define ValueFromPipelineByPropertyName = true._
+
+### Required Parameter Sets
+
+In most Azure PowerShell cmdlets, there is a bare minimum of three parameter sets that need to be implemented.
+
+#### Interactive Parameter Set
+
+This parameter set should be implemented by _every_ cmdlet - in most cases, the user provides the name of the resource that they are acting upon (`Name`) and the resource group in which they are acting in (`ResourceGroupName`).
+
+The interactive parameter set **will always be the default parameter set** for a cmdlet. This means that when PowerShell is unable to determine which parameter set a user is in, it will default to the interactive parameter set and prompt the user to provide values for the missing mandatory parameters.
```

Comment:
Probably should state that they need to explicitly set the interactive parameter set as the DefaultParameterSet.

Reply:
@maddieclayton fixed